In [51]:
import numpy as np,pandas as pd

In [52]:
p =np.genfromtxt( r'C:\Users\Sam\Downloads\Sample_data (1).csv',delimiter=',',skip_header=1)

In [53]:
p

array([[ 0., 25., 40.],
       [ 1., 30., 35.],
       [ 2., 28., 38.],
       [ 3., 32., 41.],
       [ 4., 26., 37.],
       [ 5., 29., 36.],
       [ 6., 27., 39.],
       [ 7., 33., 42.],
       [ 8., 31., 38.],
       [ 9., 28., 37.],
       [10., 30., 40.],
       [11., 29., 36.]])

# Measures of Central Tendency:

1. Calculate the mean sales for Product A and Product B. Compare the average monthly sales for the two products and discuss any observed trends.

In [63]:
x={'Product_A':[25,30,28,32,26,29,27,33,31,28,30,29],'Product_B':[40,35,38,41,37,36,39,42,38,37,40,36]}
Table=pd.DataFrame(x)
print(Table)

mean_value = np.mean(Table)

    Product_A  Product_B
0          25         40
1          30         35
2          28         38
3          32         41
4          26         37
5          29         36
6          27         39
7          33         42
8          31         38
9          28         37
10         30         40
11         29         36


In [72]:
# mean for the sale  of Product_A and Product_B
print(f'The mean for the sale of product is :',Table.mean())

The mean for the sale of product is : Product_A    29.00
Product_B    38.25
dtype: float64


#### product B's mean(38.25) is higher than product A's(29.00) mean by this we can say that product B is leading by 9.25 sales in the market as the average sales are higher

2. Find the median sales for each product and discuss how it reflects the central tendency of the sales distribution

In [81]:
sbv = sorted(Table.items())
sbv

[('Product_A',
  0     25
  1     30
  2     28
  3     32
  4     26
  5     29
  6     27
  7     33
  8     31
  9     28
  10    30
  11    29
  Name: Product_A, dtype: int64),
 ('Product_B',
  0     40
  1     35
  2     38
  3     41
  4     37
  5     36
  6     39
  7     42
  8     38
  9     37
  10    40
  11    36
  Name: Product_B, dtype: int64)]

In [83]:
x = {'Product_A': [25, 30, 28, 32, 26, 29, 27, 33, 31, 28, 30, 29],
     'Product_B': [40, 35, 38, 41, 37, 36, 39, 42, 38, 37, 40, 36]}

sortedvalues = {product: sorted(values) for product, values in x.items()}

print("Sorted Values in Ascending order :", sortedvalues)


Sorted Values in Ascending order : {'Product_A': [25, 26, 27, 28, 28, 29, 29, 30, 30, 31, 32, 33], 'Product_B': [35, 36, 36, 37, 37, 38, 38, 39, 40, 40, 41, 42]}


In [84]:
# mean for the sale  of Product_A and Product_B
print(f'The mean for the sale of product is :',Table.median())

The mean for the sale of product is : Product_A    29.0
Product_B    38.0
dtype: float64


#### after sorting the data the data set count is in even so we cans ee there are 12 values in wich the 6th one will be the median so the list of product B have the greater median most of the values are below 38 in product B and half of the distribution is greter that 38 so we can say that the product B are the outliers related to product A

3. Identify the mode of the sales data for both Product A and Product B. Discuss the significance of the mode in the context of these products.


In [96]:
def mode(data):
    frequency = {}
    for value in data:
        if value in frequency:
            frequency[value] += 1
        else:
            frequency[value] = 1
    modevalue = max(frequency, key=frequency.get)
    modefrequency = frequency[modevalue]
    return modevalue, modefrequency
x = {'Product_A': [25, 30, 28, 32, 26, 29, 27, 33, 31, 28, 30, 29],
     'Product_B': [40, 35, 38, 41, 37, 36, 39, 42, 38, 37, 40, 36]}

In [97]:
modesalesA, modefrequencyA = mode(x['Product_A'])
modesalesB, modefrequencyB = mode(x['Product_B'])
print("Mode of Product A Sales:", modesalesA, "with frequency:", modefrequencyA)
print("Mode of Product B Sales:", modesalesB, "with frequency:", modefrequencyB)


Mode of Product A Sales: 30 with frequency: 2
Mode of Product B Sales: 40 with frequency: 2


In [98]:
import statistics

In [101]:
product_A=p[:,1]
product_A

array([25., 30., 28., 32., 26., 29., 27., 33., 31., 28., 30., 29.])

In [102]:
product_B=p[:,2]
product_B

array([40., 35., 38., 41., 37., 36., 39., 42., 38., 37., 40., 36.])

In [105]:
statistics.mode(product_A)

30.0

In [106]:
statistics.mode(product_B)

40.0

#### Understanding the mode is valuable for businesses as it helps them identify patterns and make informed decisions based on the most frequently observed sales outcomes.

4. Calculate the weighted mean of the sales data, assuming each month has an equal weight. Discuss any implications of using weighted mean in this context.


In [107]:
x = {'Product_A': [25, 30, 28, 32, 26, 29, 27, 33, 31, 28, 30, 29],
     'Product_B': [40, 35, 38, 41, 37, 36, 39, 42, 38, 37, 40, 36]}

weightedmeanofA = sum(x['Product_A']) / len(x['Product_A'])
weightedmeanofB = sum(x['Product_B']) / len(x['Product_B'])

print("Weighted Mean of Product A Sales:", weightedmeanofA)
print("Weighted Mean of Product B Sales:", weightedmeanofB)


Weighted Mean of Product A Sales: 29.0
Weighted Mean of Product B Sales: 38.25


# Measures of Dispersion:

1. Compute the range of monthly sales for both Product A and Product B. Discuss how the range provides insights into the variability of sales.


In [108]:
x = {'Product_A': [25, 30, 28, 32, 26, 29, 27, 33, 31, 28, 30, 29],
     'Product_B': [40, 35, 38, 41, 37, 36, 39, 42, 38, 37, 40, 36]}
rangeA = max(x['Product_A']) - min(x['Product_A'])
rangeB = max(x['Product_B']) - min(x['Product_B'])

print("Range of Product A Sales:", rangeA)
print("Range of Product B Sales:", rangeB)


Range of Product A Sales: 8
Range of Product B Sales: 7


#### Range is calculated as the difference between the maximum and minimum sales values, providing a measure of variability.the range provides a simple measure of how much sales values vary over the observed period and will be able to understand the market and plan the sales accordingly for the future endovers,In the above problem product B has larger range representing more variability in monthly sales compared to product A

2. Calculate the variance of the sales data for both products. Discuss what the variance reveals about the spread of sales for each product.

In [110]:
np.var(Table)

Product_A    5.166667
Product_B    4.354167
dtype: float64

3. Determine the standard deviation of the sales for Product A and Product B. Discuss the significance of standard deviation in understanding the variability in sales.

In [111]:
np.std(Table)

Product_A    2.273030
Product_B    2.086664
dtype: float64

product B is with lower standard deviation , this indicates more stable and consistent sales performence with less variabilty
product A is with Higher standard deviation suggests greater variability in monthly sales, this indicates sales performence with higher variabilty unstable and inconsistent of sales

# Relative Measures of Dispersion:

1. Calculate the coefficient of variation for the sales data of Product A. Discuss how this relative measure of dispersion helps in comparing the variability of products with different average sales.

In [119]:
y=np.mean(Table)
y

33.625

In [133]:
product_A = [25, 30, 28, 32, 26, 29, 27, 33, 31, 28, 30, 29]
a= np.mean(product_A)
std_dev_A = np.std(product_A)
cv_A = (std_dev_A / a) * 100
print("Coefficient of Variation for Product A:", cv_A)

Coefficient of Variation for Product A: 7.838035458037848


2. Compute the relative range for Product B. Discuss how the relative range accounts for the scale of the sales data.

In [132]:
product_B = [40, 35, 38, 41, 37, 36, 39, 42, 38, 37, 40, 36]
range_B = max(product_B) - min(product_B)
mean_B = np.mean(product_B)
relative_range_B = (range_B / mean_B) * 100
print("Relative Range for Product B:", relative_range_B)

Relative Range for Product B: 18.30065359477124


3. Determine the relative standard deviation for Product A. Interpret the result and discuss its implications for understanding the variability relative to the mean.

In [131]:
product_A = [25, 30, 28, 32, 26, 29, 27, 33, 31, 28, 30, 29]
std_dev_A = np.std(product_A)
mean_A = np.mean(product_A)
rsd_A = (std_dev_A / mean_A) * 100
print("Relative Standard Deviation for Product A:", rsd_A)

Relative Standard Deviation for Product A: 7.838035458037848
